compiler/passes/errorHandling.cpp

```diff
@@ -136,77 +177,85 @@ bool ErrorHandlingVisitor::enterTryStmt(TryStmt* node) {
 void ErrorHandlingVisitor::exitTryStmt(TryStmt* node) {
   SET_LINENO(node);
 
-  TryInfo    info     = tryStack.top();
+  TryInfo info  = tryStack.top();
+  tryStack.pop();
+
+  TryInfo* outerTry = NULL;
+  if (!tryStack.empty())
+    outerTry = & tryStack.top();
+
   BlockStmt* tryBlock = node->body();
 
   tryBlock->insertAtHead(new DefExpr(info.errorVar));
 
   tryBlock->insertAtTail(new DefExpr(info.handlerLabel));
-  tryBlock->insertAtTail(tryHandler(node, info.errorVar));
+  tryBlock->insertAtTail(handler(node, info.errorVar, outerTry));
 
   tryBlock->remove();
   node    ->replace(tryBlock);
-  tryStack.pop();
 }
 
-AList ErrorHandlingVisitor::tryHandler(TryStmt* tryStmt, VarSymbol* errorVar) {
+AList ErrorHandlingVisitor::handler(TryStmt* tryStmt, VarSymbol* errorVar,
+                                       TryInfo* outerTry) {
```

Comment:
A nit: 'tryHandler' and 'handler' are both maybe not as good names as something like "lowerCatches" or "handleCatches" or "handleCatchesAndPropagation" etc. Change it if you like, or not...